crates/circuit/src/circuit_data.rs

```diff
@@ -1903,36 +2508,9 @@ impl CircuitData {
         &self.data
     }
 
-    /// Clone an empty CircuitData from a given reference.
-    ///
-    /// The new copy will have the global properties from the provided `CircuitData`.
-    /// The the bit data fields and interners, global phase, etc will be copied to
-    /// the new returned `CircuitData`, but the `data` field's instruction list will
-    /// be empty. This can be useful for scenarios where you want to rebuild a copy
-    /// of the circuit from a reference but insert new gates in the middle.
-    ///
-    /// # Arguments
-    ///
-    /// * other - The other `CircuitData` to clone an empty `CircuitData` from.
-    /// * capacity - The capacity for instructions to use in the output `CircuitData`
-    ///   If `None` the length of `other` will be used, if `Some` the integer
-    ///   value will be used as the capacity.
-    pub fn clone_empty_like(other: &Self, capacity: Option<usize>) -> PyResult<Self> {
-        let mut empty = CircuitData {
-            data: Vec::with_capacity(capacity.unwrap_or(other.data.len())),
-            qargs_interner: other.qargs_interner.clone(),
-            cargs_interner: other.cargs_interner.clone(),
-            qubits: other.qubits.clone(),
-            clbits: other.clbits.clone(),
-            param_table: ParameterTable::new(),
-            global_phase: Param::Float(0.0),
-            qregs: other.qregs.clone(),
-            cregs: other.cregs.clone(),
-            qubit_indices: other.qubit_indices.clone(),
-            clbit_indices: other.clbit_indices.clone(),
-        };
-        empty.set_global_phase(other.global_phase.clone())?;
-        Ok(empty)
+    /// Returns an iterator over the stored identifiers in order of insertion
+    pub fn identifiers(&self) -> impl ExactSizeIterator<Item = &CircuitIdentifierInfo> {
+        self.identifier_info.iter().map(|info| info.1)
```

Comment:
I think you should be able to use `IndexMap::values()` for this and still preserve the order.
```suggestion
        self.identifier_info.iter().values()
```

Reply:
It should actually be `identifier_info.values()`, but yes, you're right it's cleaner this way. 
Changed in d787a0e6fbc251eabbbc0cfc30b92068855fd0aa